mautrix/client/api/modules/media_repository.py

```diff
@@ -265,3 +278,12 @@ async def get_media_repo_config(self) -> MediaRepoConfig:
             return MediaRepoConfig.deserialize(content)
         except SerializerError as e:
             raise MatrixResponseError("Invalid MediaRepoConfig in response") from e
+
+    async def _upload_to_url(self, upload_url: str, post_upload_path: str, data: Any):
+        response = await self.api.session.request(Method.PUT.name, upload_url, data=data)
+        if not response.ok:
+            self.log.error(
+                f"non-ok http response from upload URL: PUT {upload_url} returned {response.status}"
+            )
+            raise Exception("non-ok http response from server")
```

Comment:
We should probably have a more specific exception here although I'm not sure what makes sense as it's out-of-homeserver/matrix context, any thoughts @tulir?

Reply:
`resp.raise_for_status()` is probably fine. It can apparently also be set as `raise_for_status=True` in the request to make it happen automatically.

This also needs its own retry loop as the standard one is only for matrix requests